hs_geographic_feature_resource/receivers.py

```diff
@@ -479,38 +542,67 @@ def geofeature_pre_add_files_to_resource(sender, **kwargs):
+                                   eastlimit=originalcoverage_dict['eastlimit'],
+                                   projection_string=originalcoverage_dict['projection_string'],
+                                   projection_name=originalcoverage_dict['projection_name'],
+                                   datum=originalcoverage_dict['datum'],
+                                   unit=originalcoverage_dict['unit'])
 
                 field_info_array = meta_dict["field_info_array"]
                 for field_info in field_info_array:
                     field_info_dict = field_info["fieldinformation"]
-                    res_obj.metadata.create_element('FieldInformation',
-                                                fieldName=field_info_dict['fieldName'],
-                                                fieldType=field_info_dict['fieldType'],
-                                                fieldTypeCode=field_info_dict['fieldTypeCode'],
-                                                fieldWidth=field_info_dict['fieldWidth'],
-                                                fieldPrecision=field_info_dict['fieldPrecision'])
+                    res_obj.metadata.\
+                        create_element('FieldInformation',
+                                       fieldName=field_info_dict['fieldName'],
+                                       fieldType=field_info_dict['fieldType'],
+                                       fieldTypeCode=field_info_dict['fieldTypeCode'],
+                                       fieldWidth=field_info_dict['fieldWidth'],
+                                       fieldPrecision=field_info_dict['fieldPrecision'])
 
                 geometryinformation_dict = meta_dict["geometryinformation"]
-                res_obj.metadata.create_element('GeometryInformation',
-                                                featureCount=geometryinformation_dict['featureCount'],
-                                                geometryType=geometryinformation_dict['geometryType']
-                                                )
+                res_obj.metadata.\
+                    create_element('GeometryInformation',
+                                   featureCount=geometryinformation_dict['featureCount'],
+                                   geometryType=geometryinformation_dict['geometryType'])
+
+                shp_xml_metadata_list = parse_shp_xml(shp_xml_full_path)
+                for shp_xml_metadata in shp_xml_metadata_list:
+                    if 'description' in shp_xml_metadata:
+                        # overwrite existing description metadata
+                        if res_obj.metadata.description:
+                            res_obj.metadata.description.delete()
+                        res_obj.metadata.create_element('description',
+                                                        abstract=shp_xml_metadata
+                                                        ['description']['abstract'])
```

Comment:
@zhiyuli I am thinking you should be able to call update_element() to achieve the metadata overwrite rather than deleting it first, then create a new element. The former should be more efficient. Checking with @pkdash to be sure here.